src/server/api/endpoints/users/recommendation.ts

```diff
@@ -15,44 +17,77 @@ export const meta = {
 };
 
 export default (params: any, me: ILocalUser) => new Promise(async (res, rej) => {
-	// Get 'limit' parameter
-	const [limit = 10, limitErr] = $.num.optional.range(1, 100).get(params.limit);
-	if (limitErr) return rej('invalid limit param');
-
-	// Get 'offset' parameter
-	const [offset = 0, offsetErr] = $.num.optional.min(0).get(params.offset);
-	if (offsetErr) return rej('invalid offset param');
-
-	// ID list of the user itself and other users who the user follows
-	const followingIds = await getFriendIds(me._id);
-
-	// ミュートしているユーザーを取得
-	const mutedUserIds = (await Mute.find({
-		muterId: me._id
-	})).map(m => m.muteeId);
-
-	const users = await User
-		.find({
-			_id: {
-				$nin: followingIds.concat(mutedUserIds)
+	var external = config.user_recommendation.external
+
+	if (external) {
+		var userName = me.username
+		var hostName = config.hostname
+		var limit = params.limit
+		var offset = params.offset
+		var timeout = config.user_recommendation.timeout
+		var engine = config.user_recommendation.engine
+		var url
+		url = engine
+		url = url.replace('{{host}}', hostName)
+		url = url.replace('{{user}}', userName)
+		url = url.replace('{{limit}}', limit)
+		url = url.replace('{{offset}}', offset)
+		request(
+			{
+				url: url,
+				timeout: timeout,
+				json: true,
+				followRedirect: true,
+				followAllRedirects: true
 			},
-			isLocked: false,
-			$or: [{
-				lastUsedAt: {
-					$gte: new Date(Date.now() - ms('7days'))
+			function (error: any, response: any, body: any) {
```

Comment:
You should use arrow expression instead of function expression.

*Existing Codes Example:*

https://github.com/syuilo/misskey/blob/ca0b56ee5794ad996badbaeef0c54dfb0b4b1efe/src/server/api/service/github.ts#L68-L71

Reply:
Done. Thanks.